fdbclient/include/fdbclient/DatabaseContext.h

```diff
@@ -156,6 +156,8 @@ class WatchMetadata : public ReferenceCounted<WatchMetadata> {
 
 	WatchMetadata(Reference<const WatchParameters> parameters)
 	  : watchFuture(watchPromise.getFuture()), parameters(parameters) {}
+
+	~WatchMetadata() { watchFutureSS.cancel(); }
```

Comment:
Is this needed? This should only change the behavior if someone else is holding the future.

Reply:
No. It is unnecessary. Thought it was necessary though.